client/blocks/comment-detail/index.jsx

```diff
@@ -400,8 +404,11 @@ const mapStateToProps = ( state, ownProps ) => {
 
 	const postId = get( comment, 'post.ID' );
 
-	// TODO: eventually it will be returned already decoded from the data layer.
-	const postTitle = decodeEntities( get( comment, 'post.title' ) );
+	const post = getSitePost( state, siteId, postId );
+
+	const postTitle =
+		decodeEntities( get( comment, 'post.title' ) ) ||
+		decodeEntities( stripHTML( get( post, 'excerpt' ) ) );
```

Comment:
not a big deal, but if we have unicode whitespace characters we'll still get a visual blank. the RegExp `\s` though I believe takes those into consideration, so if we wanted to iterate…

```js
const getTitle = ( post, comment ) => {
	const ws = /^\s*$/;
	const title = decodeEntities( get( comment, 'post.title' ) );
	if ( ! ws.test( title ) ) {
		return title;
	}

	// might `post` be `undefined`?
	// if not, we can just use `post.excerpt`
	// and leave out the `get()`
	const excerpt = stripHTML( get( post, 'excerpt' ) );
	if ( ws.test( excerpt ) ) {
		return excerpt;
	}

	return translate( 'Untitled Post' );
}
```

Reply:
Good catch, I'll add an issue about this.